src/libxtp/activedensitymatrix.cc

```diff
@@ -0,0 +1,68 @@
+/*
+ *            Copyright 2009-2020 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "votca/xtp/activedensitymatrix.h"
+#include "votca/xtp/aomatrix.h"
+#include <votca/tools/eigenio_matrixmarket.h>
+
+namespace votca {
+namespace xtp {
+
+Eigen::MatrixXd ActiveDensityMatrix::compute_Dmat_A() {
+  Eigen::MatrixXd new_mo_coeff = orbitals.getPMLocalizedOrbitals();
+  Eigen::MatrixXd dmat = activedensitymatrix(new_mo_coeff);
+  // votca::tools::EigenIO_MatrixMarket::WriteMatrix("ch3oh.mm", dmat);
+  return dmat;
+}
+
+Eigen::MatrixXd ActiveDensityMatrix::activedensitymatrix(
+    Eigen::MatrixXd &new_mo_coeff) {
+  QMMolecule mol = orbitals.QMAtoms();
+  basis.Load(orbitals.getDFTbasisName());
+  aobasis.Fill(basis, mol);
+  AOOverlap overlap;
+  overlap.Fill(aobasis);
```

Comment:
you only need them locally please remove member variables and make them local variables 

in simple words remove them from the header and declare them here

Reply:
Done